compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1793,6 +1794,30 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                             );
                         }
                     }
+
+                    // if the attribute is malformed, it may return None from this
+                    // but an error will have already been emitted, so this code should just skip such attributes
+                    if let Some((
+                        _,
+                        MetaItemLit {
+                            kind: ast::LitKind::Int(literal, ast::LitIntType::Unsuffixed),
+                            ..
+                        },
+                    )) = hint.singleton_lit_list()
+                    {
+                        let val = literal.get() as u64;
+                        // ignore values greater than 2^29, a different error will be emitted
+                        if val <= 2_u64.pow(29) {
+                            let max = Size::from_bits(self.tcx.sess.target.pointer_width)
+                                .signed_int_max() as u64;
+                            if val > max {
+                                self.dcx().emit_err(errors::InvalidReprAlignForTarget {
+                                    span: hint.span(),
+                                    size: max,
+                                });
+                            }
+                        }
+                    }
                 }
```

Comment:
Suggestion: can you please extract this into a helper fn? We should keep the top-level check attr visitor more of an "orchestrator" and delegate non-trivial check details to helpers.